packages/web/src/components/track/DownloadSection.tsx

```diff
@@ -0,0 +1,137 @@
+import { useCallback, useState } from 'react'
+
+import { useCurrentStems, ID } from '@audius/common'
+import {
+  Flex,
+  Box,
+  Text,
+  IconReceive,
+  Button,
+  ButtonType,
+  ButtonSize,
+  IconCaretDown
+} from '@audius/harmony'
+import { SegmentedControl } from '@audius/stems'
+
+import { Icon } from 'components/Icon'
+import { Expandable } from 'components/expandable/Expandable'
+
+import { DownloadRow } from './DownloadRow'
+
+const messages = {
+  title: 'Stems & Downloads',
+  choose: 'Choose File Quality',
+  mp3: 'MP3',
+  original: 'Original',
+  downloadAll: 'Download All'
+}
+
+enum Quality {
+  MP3 = 'MP3',
+  ORIGINAL = 'ORIGINAL'
+}
+
+type DownloadSectionProps = {
+  trackId: ID
+  onDownload: (trackId: ID, category?: string, parentTrackId?: ID) => void
+}
+
+export const DownloadSection = ({
+  trackId,
+  onDownload
+}: DownloadSectionProps) => {
+  const { stemTracks } = useCurrentStems({ trackId })
+  const shouldDisplayDownloadAll = stemTracks.length > 1
```

Comment:
what if parent track is downloadable and `stemTracks.length == 1`?

Reply:
ah good catch there are separate cases for whether the original quality is available. i thought this was only for "download all" button on the last row of the table.